metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java

```diff
@@ -272,14 +277,50 @@ void getCreatePublicationRequestReturnsContributorWhenInputIsValidContributor(Li
         assertThat(actual, containsInAnyOrder(expected));
     }
 
+    @Test
+    void getCreatePublicationRequestReturnsTitleWhenInputContainsOnlyHtmlHeadTitle() throws IOException {
+        CreatePublicationRequest request = getCreatePublicationRequest(FAKE_TITLE);
+        String actual = request.getEntityDescription().getMainTitle();
+        assertThat(actual, equalTo(FAKE_TITLE));
+    }
+
+    @ParameterizedTest(name = "MetaTag {0} is assessed as title")
+    @ArgumentsSource(MetaTagTitleProvider.class)
+    void getCreatePublicationRequestReturnsTitleWhenInputContainsMetaTitle(String metaTagName, String metaTagContent)
+            throws IOException, InterruptedException {
+        CreatePublicationRequest request =
+                getCreatePublicationRequest(metaTagName, metaTagContent);
+        String actual = request.getEntityDescription().getMainTitle();
+        assertThat(actual, equalTo(metaTagContent));
+    }
+
+    @Test
+    void getCreatePublicationRequestReturnsLongestTitleWhenInputContainsMultipleTitles() throws IOException {
+        List<MetaTagPair> metaTags = List.of(new MetaTagPair(DC_TITLE, "a short title"),
```

Comment:
The code works correctly, but this test would pass if the code just favored the html title. Maybe one more test where the longest title is in the meta tags?